iguana.corecontroller/src/main/java/org/aksw/iguana/cc/worker/AbstractWorker.java

```diff
@@ -89,18 +89,28 @@ public abstract class AbstractWorker implements Worker {
 	protected int queryHash;
 
 	public AbstractWorker(String taskID, Connection connection, String queriesFile, @Nullable Integer timeOut, @Nullable Integer timeLimit, @Nullable Integer fixedLatency, @Nullable Integer gaussianLatency, String workerType, Integer workerID) {
-		this.taskID=taskID;
+		this.taskID = taskID;
 		this.workerID = workerID;
-		this.workerType = workerType;
+
+		if (workerType != null) {
```

Comment:
Very nice solution. Please double a final check if the constructor parameter `workerType` is ever used in a meaningful way. Otherwise, please remove it and let this Code snipped handle setting `this.workerType`.